### PR TITLE
[tiptap-editor] improve: Ricos Extension Types

### PR DESCRIPTION
**Pull request by Barackos**
- Improvement of RicosExtensions' types
- Circular dependency fix

### CODE REVIEW

File: packages/common/web/src/types/tiptapTypes.ts [hunk 1]
```diff
@@ -1,13 +1,35 @@
 import * as ricosSchema from 'ricos-schema';
 import { NodeConfig, MarkConfig, ExtensionConfig, mergeAttributes } from '@tiptap/core';
 export type TiptapExtensionConfig = NodeConfig | MarkConfig | ExtensionConfig;
-export interface RicosTiptapExtension<T extends TiptapExtensionConfig> {
-  type: 'node' | 'mark' | 'extension';
-  Component?: React.ComponentType;
+export interface ExtensionTypeToConfig {
```
[the remark this answers is not in the document]
Comment (lxgreen) [reply packages/common/web/src/types/tiptapTypes.ts:4]:
```
CreateExtension<P> = (params: P) => RicosTiptapExtension
```

**Review by lxgreen — CHANGES_REQUESTED** [review]

File: packages/editor-common/web/src/Utils/tiptap/types.ts [hunk 2]
```diff
@@ -1,8 +1,7 @@
...
+import { RicosTiptapExtension, RicosExtension } from 'wix-rich-content-common';
 
-export interface CreateExtensionParams<T extends TiptapExtensionConfig>
-  extends Omit<RicosTiptapExtension<T>, 'type'> {}
+export type CreateExtensionParams<T extends RicosExtension> = Omit<RicosTiptapExtension<T>, 'type'>;
 
-export type CreateExtension<T extends TiptapExtensionConfig> = (
+export type CreateExtension<T extends RicosExtension> = (
   params: CreateExtensionParams<T>
 ) => RicosTiptapExtension<T>;
```
Comment (mharsat) [inline packages/editor-common/web/src/Utils/tiptap/types.ts:7]: `RicosTiptapExtension` is unnecessary

```suggestion
export type CreateExtension<T extends RicosExtension> = (
  params: CreateExtensionParams<T>
) => T;
```

File: packages/editor-common/web/src/Utils/tiptap/extensionCreators.ts [hunk 3]
```diff
@@ -1,25 +1,30 @@
...
+  RicosMarkExtension,
+  RicosNodeExtension,
+} from 'wix-rich-content-common';
 import { CreateExtension } from './types';
-import { NodeConfig, MarkConfig, ExtensionConfig } from '@tiptap/core';
+import { ExtensionConfig } from '@tiptap/core';
 import { PluginProps } from '../..';
 
-export interface RicosExtensionConfig extends ExtensionConfig {
+export interface RicosGenericExtensionConfig extends ExtensionConfig {
```
Comment (mharsat) [inline packages/editor-common/web/src/Utils/tiptap/extensionCreators.ts:10]: Here too, avoid using generic, it doesn't represent what you're looking for and is confusing in the context of typescript

**Comment by Barackos** [comment]
Implemented differently in #2833. Closing

**Pull request closed without merging**